lightning_examples/augmentations/data_augmentation_kornia_lightning_gpu.py

```diff
@@ -0,0 +1,232 @@
+# -*- coding: utf-8 -*-
+# ---
+# jupyter:
+#   jupytext:
+#     formats: ipynb,py:percent
+#     text_representation:
+#       extension: .py
+#       format_name: percent
+#       format_version: '1.3'
+#       jupytext_version: 1.11.3
+#   kernelspec:
+#     display_name: Python 3
+#     language: python
+#     name: python3
+# ---
+
+# %% [markdown] colab_type="text" id="x83-rnVKT8Wo"
+# # GPU and batched data augmentation with Kornia and PyTorch-Lightning
+#
+# **NOTE:** Adaptation of the original post found in [Kornia tutorials](https://kornia-tutorials.readthedocs.io/en/latest/).
+
+# %% [markdown] colab_type="text" id="iCsre0XmawoR"
+# ## Setup
+# We first need to install Kornia and PyTorch Lightning and for convenience also torchmetrics.
+
+# %% colab={} colab_type="code" id="dEeUzX_5aLrX"
+# ! pip install kornia pytorch_lightning torchmetrics -qU
+
+# %% colab={} colab_type="code" id="Z_XTj-y1gYJL"
+import os
+
+import kornia as K
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import pytorch_lightning as pl
+import torch
+import torch.nn as nn
+import torchmetrics
+import torchvision
+from pytorch_lightning import Trainer
+from torch.nn import functional as F
+from torch.utils.data import DataLoader
+from torchvision.datasets import CIFAR10
+
+AVAIL_GPUS = 1 if torch.cuda.is_available() else None
```

Comment:
```suggestion
AVAIL_GPUS = min(1, torch.cuda.device_count())
```

Reply:
this made the cpu to hang

Reply:
is this linked to PL or Kornia usage? bc it forks fine on other notebooks

Reply:
we can try again but it wasn't working yesterday